ts/features/fci/screens/valid/FciDocumentsScreen.tsx

```diff
@@ -23,16 +38,192 @@ const styles = StyleSheet.create({
+            setSignaturePage(page + 1);
+            // The signature field is extracted by its unique_name.
+            // Using low-level acrofield (acrobat field) it is possible
+            // to obtain the elements of the signature field such as the
+            // box that contains it. Once the box is obtained, its
+            // coordinates are used to draw a rectangle on the related page.
+            const signature = res.getForm().getSignature(uniqueName);
+            const [widget] = signature.acroField.getWidgets();
+            const rect = widget.getRectangle();
+            res.getPage(page).drawRectangle({
+              x: rect.x,
+              y: rect.y,
+              width: rect.width,
+              height: rect.height,
+              color: rgb(0, 0.77, 0.79),
+              opacity: 0.5,
+              borderOpacity: 0.75
+            });
+          })
+        );
+
+        return res.saveAsBase64().then(r => setPdfString(pdfFromBase64(r)));
+      });
+    },
+    [documents, currentDoc]
+  );
+
+  /**
+   * Get the pdf url from documents,
+   * download it as base64 string and
+   * load the pdf as pdf-lib object
+   * to draw a rect over the signature field
+   * giving a set of coordinates
+   * @param attrs the signature field attrs containing the coords
+   */
+  const drawRectangleOverSignatureFieldByCoordinates = React.useCallback(
+    async (attrs: SignatureFieldToBeCreatedAttrs) => {
+      // TODO: refactor this function to use fp-ts
+      const doc = documents[currentDoc];
+      const url = doc.url;
+
+      const existingPdfBytes = await ReactNativeBlobUtil.fetch("GET", url).then(
+        res => res.base64()
+      );
+
+      await PDFDocument.load(pdfFromBase64(existingPdfBytes)).then(res => {
+        const page = attrs.page;
+        setSignaturePage(page);
+        // The signature box is drawn using the coordinates of the signature field.
+        res.getPage(page).drawRectangle({
+          x: attrs.coordinates.top_right.x,
+          y: attrs.coordinates.top_right.y,
+          width: attrs.coordinates.bottom_left.x,
+          height: attrs.coordinates.bottom_left.y,
+          color: rgb(0, 0.77, 0.79),
+          opacity: 0.5,
+          borderOpacity: 0.75
+        });
+
+        return res.saveAsBase64().then(r => setPdfString(pdfFromBase64(r)));
+      });
+    },
+    [documents, currentDoc]
+  );
 
-  // TODO: abort signature flow
-  const onCancelPress = () => constNull;
+  const onSignatureDetail = React.useCallback(
+    (attrs: ExistingSignatureFieldAttrs | SignatureFieldToBeCreatedAttrs) => {
+      if (hasUniqueName(attrs)) {
+        drawRectangleOverSignatureFieldById(attrs.unique_name).catch(
+          readableReport // TODO: it should be displayed to the user?
+        );
+      } else {
+        drawRectangleOverSignatureFieldByCoordinates(attrs).catch(
+          readableReport // TODO: it should be displayed to the user?
+        );
```

Comment:
@gunzip I have added a TODO to catch this in future refinement. What do you think?

Reply:
TODOs without a linked (Jira) issue become DONTDO :-)